sphinx/domains/std.py

```diff
@@ -538,6 +540,23 @@ def __init__(self, env: "BuildEnvironment") -> None:
     def objects(self) -> Dict[Tuple[str, str], Tuple[str, str]]:
         return self.data.setdefault('objects', {})  # (objtype, name) -> docname, labelid
 
+    def note_object(self, objtype: str, name: str, labelid: str, location: Any = None
+                    ) -> None:
```

Comment:
I added a new helper method to register an object to std domain. This does not provides `docname` argument like `add_object()`. I think it was wrong design. Extensions can register objects only during processing the document. So no `docname` argument is needed.